app/Phragile/ActionHandler/SprintStoreActionHandler.php

```diff
@@ -0,0 +1,125 @@
+<?php
+namespace Phragile\ActionHandler;
+
+use Phragile\PhabricatorAPI;
+use \Flash;
+use Illuminate\Support\Facades\Redirect;
+
+class SprintStoreActionHandler {
+	private $userPhabricatorAPI;
+	private $botPhabricatorAPI;
+	private $redirect;
+	private $sprint;
+	private $user;
+
+	public function __construct(PhabricatorAPI $userPhabricatorAPI, PhabricatorAPI $botPhabricatorAPI)
+	{
+		$this->userPhabricatorAPI = $userPhabricatorAPI;
+		$this->botPhabricatorAPI = $botPhabricatorAPI;
+	}
+
+	public function performAction(\Sprint $sprint, \User $user)
+	{
+		$this->sprint = $sprint;
+		$this->user = $user;
+
+		$this->validate();
+		$this->connectUserToPhabricator();
+		$this->createCorrespondingPhabricatorProject();
+		$this->save();
+	}
+
+	private function validate()
+	{
+		if ($this->previousActionFailed()) return;
+
+		$validation = $this->sprint->validate();
+		if ($validation->fails())
+		{
+			Flash::error(implode(' ', $validation->messages()->all()));
+			$this->redirect = Redirect::back();
+		}
+	}
+
+	private function connectUserToPhabricator()
+	{
+		if ($this->previousActionFailed()) null;
+
+		try
+		{
+			$this->userPhabricatorAPI->connect($this->user->username, $this->user->conduit_certificate);
+		} catch(\ConduitClientException $e)
+		{
+			$this->redirectBackWithError($e->getMessage());
+		}
+	}
+
+	private function createCorrespondingPhabricatorProject()
+	{
+		if ($this->previousActionFailed()) return;
+
+		try
+		{
+			$phabricatorProject = $this->userPhabricatorAPI->createProject($this->sprint->title, [$this->user->phid]);
+			$this->sprint->connectWithPhabricatorProject($phabricatorProject);
+		} catch(\ConduitClientException $e)
+		{
+			$this->connectIfPhabricatorProjectExists($e->getMessage());
+		}
+	}
+
+	private function connectIfPhabricatorProjectExists($errorMessage)
+	{
+		if (str_contains($errorMessage, 'Project name is already used'))
+		{
+			$this->connectWithPhabricatorProject();
+		} else $this->redirectBackWithError('Could not create a Phabricator project for this sprint.');
+	}
+
+	private function connectWithPhabricatorProject()
+	{
+		$this->sprint->connectWithPhabricatorProject($this->fetchPhabricatorProject());
```

Comment:
I'm very dubious about using the no-curly-bracket notation for if and if-else statements. For the guard clauses above it's not that bad. However here I'd definitely use them and put the bodies of the thing on their own line.